src/map/utils/trustutils.cpp

```diff
@@ -344,18 +349,26 @@ namespace trustutils
         auto adjustedDamage   = baseDamage * damageMultiplier;
         auto finalDamage      = std::max(adjustedDamage, 1.0f);
 
-        (dynamic_cast<CItemWeapon*>(PTrust->m_Weapons[SLOT_MAIN]))->setDamage(static_cast<uint16>(finalDamage));
-        (dynamic_cast<CItemWeapon*>(PTrust->m_Weapons[SLOT_RANGED]))->setDamage(static_cast<uint16>(finalDamage));
-        (dynamic_cast<CItemWeapon*>(PTrust->m_Weapons[SLOT_AMMO]))->setDamage(static_cast<uint16>(finalDamage));
+        // Trust do not really have weapons, but they are modelled internally as
+        // if they do.
+        auto* mainWeapon   = (CItemWeapon*)PTrust->m_Weapons[SLOT_MAIN];
```

Comment:
Is there a reason for the C-style casts here?

Reply:
Also, is there a chance that any of these weapon pointers could be null?

Reply:
Just while I copy/paste to get it working. They've never failed _yet_ with brutal casting